pkg/providers/gitlab.go

```diff
@@ -60,8 +64,61 @@ func (g *gitLab) Fetch() (*File, error) {
 	}
 
 	candidates := []*assets.Asset{}
+	candidateURLs := map[string]struct{}{}
+
+	packages, _, err := g.client.Packages.ListProjectPackages(projectPath, &gitlab.ListProjectPackagesOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, v := range packages {
+		if fmt.Sprintf("v%s", v.Version) == release.TagName {
+			packageFiles, _, err := g.client.Packages.ListPackageFiles(projectPath, v.ID, &gitlab.ListPackageFilesOptions{})
+			if err != nil {
+				return nil, err
+			}
+			for _, f := range packageFiles {
+				assetURL := fmt.Sprintf("%sprojects/%s/packages/%s/%s/%s/%s",
+					g.client.BaseURL().String(),
+					url.PathEscape(projectPath),
+					v.PackageType,
+					v.Name,
+					v.Version,
+					f.FileName,
+				)
+				if _, exists := candidateURLs[assetURL]; !exists {
+					asset := &assets.Asset{
+						Name:        v.Name,
+						DisplayName: fmt.Sprintf("%s (%s package)", f.FileName, v.PackageType),
+						URL:         assetURL,
+					}
+					candidates = append(candidates, asset)
+					log.Debugf("Adding %s with URL %s", asset.Name, asset.URL)
+				}
+				candidateURLs[assetURL] = struct{}{}
+			}
+		}
+	}
+
+	project, _, err := g.client.Projects.GetProject(projectPath, &gitlab.GetProjectOptions{})
+	if err != nil {
+		return nil, err
+	}
+	projectUploadsURL := fmt.Sprintf("%s/uploads/", project.WebURL)
+	projectIsPublic := project.Visibility == gitlab.PublicVisibility
```

Comment:
Hey Sune, I was trying this PR locally in my machine using this repo: gitlab.com/graphviz/graphviz and seems like the `project.Visibility` variable comes empty. However, `project.Public` is set correctly. 

Shall we change this?

Reply:
I will test today and see if Public gives the right result for private projects, too. Otherwise I'll make a fallback check so we try both.

Reply:
When not authenticated with a personal access token, I don't get the `"visibility"` or `"public"` fields in the API response for a public project. However, with a token towards a public project such as `gitlab.com/graphviz/graphviz`, I get `"visibility": public` but still no `"public"` field, which makes me think it's a deprecated field left around in xanzy/go-gitlab.

So I'm inclined to determine if a project is public like so, since I've now copied the token to the `type gitLab struct`:

```go
projectIsPublic := g.token == "" || project.Visibility == gitlab.PublicVisibility
```

Reply:
> So I'm inclined to determine if a project is public like so, since I've now copied the token to the `type gitLab struct`:

SGTM!